CHANGELOG.md

```diff
@@ -1,3 +1,21 @@
+## 0.13.0 (July 08, 2024)
+
+NOTES:
+* **This release requires juju controller version 2.9.49 or later juju.**
+* This release uses juju client api code from the juju 3.5.1 release.
+
+ENHANCEMENTS:
+
+* Bug reports are enhanced to require more information, including the Juju controller version.
+* Support for application resources to use storage is added. You can now use `storage` and `storage_directives` fields in an application resource to utilize storage in your application.
+* [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) are now required in contributions to the repository.
+* Provider test runs are improved to run faster.
+
+BUG FIXES:
+
+* Placement and constraints directives are allowed to co-exist in machine resources. Added by @hmlanigan in https://github.com/juju/terraform-provider-juju/pull/499
+* Auto discovery of data necessary to connect to the juju controller is fixed by @hmlanigan in https://github.com/juju/terraform-provider-juju/pull/503
```

Comment:
Not worth including - this is a fix to something which was landed during this development cycle and didn't make it to the field.